docarray/index/backends/elastic.py

```diff
@@ -104,29 +107,36 @@ def __init__(self, db_config=None, **kwargs):
                 continue
 
             mappings['properties'][col_name] = self._create_index_mapping(col)
+            self._logger.debug(f'Index mapping created for column {col_name}')
 
         if self._client.indices.exists(index=self.index_name):
             self._client_put_mapping(mappings)
+            self._logger.debug(f'Put mapping for index {self.index_name}')
         else:
             self._client_create(mappings)
+            self._logger.debug(f'Created new index {self.index_name} with mappings')
 
         if len(self._db_config.index_settings):
             self._client_put_settings(self._db_config.index_settings)
+            self._logger.debug('Updated index settings')
 
         self._refresh(self.index_name)
+        self._logger.debug(f'Refreshed index {self.index_name}')
 
     @property
     def index_name(self):
         default_index_name = (
             self._schema.__name__.lower() if self._schema is not None else None
         )
         if default_index_name is None:
-            raise ValueError(
+            self._logger.error(
                 'A ElasticDocIndex must be typed with a Document type.'
                 'To do so, use the syntax: ElasticDocIndex[DocumentType]'
             )
-
-        return self._db_config.index_name or default_index_name
+            raise ValueError
+        index_name = self._db_config.index_name or default_index_name
```

Comment:
You can do smth like


error_msg = 'A ElasticDocIndex must be typed with a Document type.'
                'To do so, use the syntax: ElasticDocIndex[DocumentType]'

self._logger.error(error_msg)
raise ValueError(error_msg)

Reply:
oh okay.